Automattic/jetpack
# VideoPress: Update format status checks to support removal or addition of more transcode formats

Upcoming changes to VideoPress required some changes to the way Jetpack checks if a transcode has finished. In fact, this was bit buggy already because the code was checking that _every_ format was finished which could not be the case for every video uploaded. (For example, if a video is longer than 2 hours currently we don't transcode it to `dvd` or `hd` formats) We would also like to remove `ogg` as a format soon and this fix was required to support that as well.

#### Changes proposed in this Pull Request:
* This PR updates the `videopress_is_finished_processing()` function to check for the `finished` boolean in the api response, which is set after a video is ready to be played. I've also included a change to the XML-RPC response handler so that it will store any format status update (like `hd_1080p` which will be added in production soon).

#### Jetpack product discussion
pxWta-WR-p2

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
* Apply PR and ensure your site has a plan with VideoPress enabled.
* Head to the post editor and upload a new video using the video block. It should upload and transcode normally showing the converting screen, etc.
* Also try uploading a video to `/wp-admin/upload.php`. You should see the conversion placeholder image. Refreshing the page after the conversion finishes should show the video thumbnail, and clicking on it should load the video details normally.
* If you want to test that Jetpack stores the new `hd_1080p` format status, add the 'JETPACK__SANDBOX_DOMAIN' constant to your dev site (instructions in field guide), and sandbox transcodes on your wp.com sandbox (ping me if you need instructions, make sure to add `attempt_transcode_video` to your sandboxed job list). After uploading a video, you should see `hd_1080p` status in the `files_statuses` meta array. You can also test removal of ogg this way as well.

#### Proposed changelog entry for your changes:


* No changelog entry needed.
